sed/loader/flash/loader.py

```diff
@@ -723,20 +735,143 @@ def read_dataframe(
             ],
         )
         dataframe = dataframe.dropna(
-            subset=self.get_channels_by_format("per_electron"),
+            subset=self.get_channels_by_format(["per_electron"]),
         )
 
         if collect_metadata:
             metadata_retriever = MetadataRetriever(self._config["metadata"])
             metadata = metadata_retriever.get_metadata(
                 beamtime_id=self._config["dataframe"]["beamtime_id"],
-                runs=runs,
+                runs=list(runs),
                 metadata=self.metadata,
             )
         else:
             metadata = self.metadata
 
         return dataframe, metadata
 
+    def get_files_from_run_id(
+        self,
+        run_id: str,
+        folders: Union[str, Sequence[str]] = None,
+        extension: str = "h5",
+        **kwds,
+    ) -> List[str]:
+        """Returns a list of filenames for a given run located in the specified directory
+        for the specified data acquisition (daq).
+
+        Args:
+            run_id (str): The run identifier to locate.
```

Comment:
I dont understand the switch from runs to run_id, and also making it a str, when it is always a int and would preferably be a sequence of ints

Reply:
Well, your original function had run_number here. Your implementation between base loader and flash loader was insofar inconsistent, as flash loader defined runs as Sequence[int], and base loader as Sequence[str]. I thought the second one as the more flexible, as it allows, e.g., also to put alphanumeric hashes or so as run identifiers. I changed this throughout and made it consistent. If you want, you can rename _run_id_ here to _run_, but then do it in all the loaders. As this function is called sequentially for all elements in runs, it takes only one element of the content of runs.

Reply:
Yes I realize the inconsistency you mention, but it was purposefully overloaded, as there is no point for flash users to always put quotes for runs. But a type hint is just a type hint and a sequence of ints should also work if we just include a line with str(runs)

Reply:
Mypy checks typing definitions for consistency, pylint also the consistency of subclass defitinitions. If you want your code to be consistent and these tools to pass the tests, you need to be consistent in your definitions. 
Practically, Python does not check the type on execution, and your code (f-string) also works with int. I added an option now such that you also can use a single int or list of ints now, but adding these also to the Union will make the thing look more bulky.